conans/client/graph/graph_manager.py

```diff
@@ -57,15 +56,28 @@ def load_consumer_conanfile(self, conanfile_path, info_folder, output,
                                 deps_info_required=False):
         """loads a conanfile for local flow: source, imports, package, build
         """
-        profile = read_conaninfo_profile(info_folder) or self._client_cache.default_profile
-        cache_settings = self._client_cache.settings.copy()
-        cache_settings.values = profile.settings_values
-        # We are recovering state from captured profile from conaninfo, remove not defined
-        cache_settings.remove_undefined()
-        processed_profile = ProcessedProfile(cache_settings, profile, None)
+        try:
+            graph_info = GraphInfo.load(info_folder)
+            profile = graph_info.profile
+            profile.process_settings(self._client_cache, preprocess=False)
+            # This is the hack of recovering the options from the graph_info
+            profile.options.update(graph_info.options)
+        except Exception:
```

Comment:
Could this exception or code to in the "try" be narrowed? There is a lot of things it could be hiding.

Reply:
:+1: